aea/aea_builder.py

```diff
@@ -819,16 +817,36 @@ def build(
             decision_maker_handler_class=self._get_decision_maker_handler_class(),
             skill_exception_policy=self._get_skill_exception_policy(),
             default_routing=self._get_default_routing(),
+            default_connection=self._get_default_connection(),
             loop_mode=self._get_loop_mode(),
             runtime_mode=self._get_runtime_mode(),
             **deepcopy(self._context_namespace),
         )
         # load connection
-        self._load_and_add_connections(aea, wallet, connection_ids=connection_ids)
-        aea.multiplexer.default_routing = self._get_default_routing()
-        self._load_and_add_skills(aea.context, resources)
+        # TODO check address is the same of identity address
+        # TODO filter connections with connection_ids
+        self._load_and_add_components(
+            ComponentType.CONNECTION,
+            resources,
+            identity=identity,
+            crypto_store=wallet.connection_cryptos,
+        )
+        connection_ids = self._process_connection_ids(connection_ids)
+        self._unregister_non_selected_connections(resources, connection_ids)
```

Comment:
Should not happen here I think...feels wrong

Reply:
I agree. We should add a "connection filter" so that we retrieve only the active connection. Again, that's for a later PR.